awslogs/core.py

```diff
@@ -175,8 +175,9 @@ def consumer():
                     )
 
                 message = event['message']
-                if self.query is not None and message[0] == '{':
-                    parsed = json.loads(event['message'])
+                json_start = message.find('{')
+                if self.query is not None and json_start != -1:
+                    parsed = json.loads(message[json_start:])
```

Comment:
I think this is a little more optimistic than it was before, and just as far from robust. Is there a reasonable solution to stop possibly unexpected messages from flowing up the script? (I'm assuming it blows up currently).

_Tangent/something of curiosity for me:_ could this code/processing be offloaded into boto3/AWS by using [filter_log_events(…)](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/logs.html#CloudWatchLogs.Client.filter_log_events)?

Reply:
I agree with @Code0x58 , this is way too optimistic.

In terms of offloading this to aws, the only option they provide to filter (afaik) is already present in ``--filter-pattern``, anything els e hast to be done in the client.